scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -688,7 +690,20 @@ object TreeSyntax {
           s(p(AnyInfixTyp, t.tpe), " ", kw("match"), " {", r(t.cases.map(i(_)), ""), n("}"))
         )
       case t: Type.Placeholder =>
-        if (dialect.allowQuestionMarkPlaceholder) m(SimpleTyp, s(kw("?"), t.bounds))
+        /* In order not to break existing tools `.syntax` should still return
+         * `_` instead `?` unless specifically used.
+         */
+        def isScala3Dialect = dialect.allowSignificantIndentation
+        def questionMarkUsed = t.origin match {
```

Comment:
Alternatively, to allow parsing old scala211 and printing new scala3:
```
def forceUnderscore = t.origin match {
  case o: Origin.Parsed => // could use `?` but didn't
    o.dialect.allowQuestionMarkPlaceholder &&
    t.tokens.headOption.exists(_.is[Token.Underscore])
  case _ => true
}
```

Reply:
Right, I haven't thought about this flow. Should be ok now.